tests/utils_tests/test_lazyobject.py

```diff
@@ -32,6 +32,29 @@ def _setup(self):
 
         return AdHocLazyObject()
 
+    def test_getattribute(self):
+        """
+        Proxy methods don't exist on wrapped objects unless they're set.
+        """
+        attrs = [
+            "__getitem__",
+            "__setitem__",
+            "__delitem__",
+            "__iter__",
+            "__len__",
+            "__contains__",
+        ]
+        foo = Foo()
+
+        obj = self.lazy_wrap(foo)
+        for attr in attrs:
+            with self.subTest(attr):
+                self.assertFalse(hasattr(obj, attr))
+
+            with self.subTest(attr):
+                setattr(foo, attr, attr)
+                self.assertEqual(getattr(foo, attr), attr)
```

Comment:
This should be called on re-wrapped `obj`.

Reply:
You are right. Thanks for fixing it 🙌